### DISCUSSION TITLE
Create files and folders from the file explorer side bar.

### DISCUSSION THREAD

**Discussion by mattw-webcv** [lapce/lapce#3100] [Ideas]
Hello, I'm new to the entirety of the git/github scene, please be kind!

In short, from the file explorer view- right click on any opened folder to contextually create a new file/folder and name it.
If this is already in the roadmap, or other ideas/discussion topics I've not yet found it mentioned.

**Comment by mattw-webcv** [comment]
> Hello, I'm new to the entirety of the git/github scene, please be kind!
> 
> In short, from the file explorer view- right click on any opened folder to contextually create a new file/folder and name it. If this is already in the roadmap, or other ideas/discussion topics I've not yet found it mentioned.

I saw in discord this has already been implemented. So I'll change my question.
As a Fedora 39 user, is there an easy and straight forward way to install latest nightly version?

**Comment by MinusGix** [comment]
There's https://copr.fedorainfracloud.org/coprs/titaniumtown/lapce/ but it looks like the last build was done a month ago so its out of date.  
So it seems like the best way is to just download the nightly binary and run it.

**Reply by mattw-webcv** [reply]
Thank you for the reply, however I looked at the nightly releases; and so long as I'm not mistaken, there isn't a fedora version.
Completely understandable, but still a little upsetting.
https://github.com/lapce/lapce/releases

**Reply by panekj** [reply]
There is standard Linux build which MinusGix meant: https://github.com/lapce/lapce/releases/download/nightly/lapce-linux-amd64.tar.gz

**Reply by mattw-webcv** [reply]
Thanks panekj but I already found it, and it doesn't launch.
I hope the following is somewhat helpful.

The only 2 errors I could get through bash are:
1:
"./lapce: error while loading shared libraries: libssl.so.1.1: cannot open shared object file: No such file or directory"

2:
"bash lapce
lapce: lapce: cannot execute binary file"

I've tried the following methods:
Nautilus/file explorer:
properties + make executable

CLI:
lapce
./lapce
bash lapce
bash ./lapce

**Reply by MinusGix** [reply]
The first sounds like you need to install openssl1.1 through your package manager. (`sudo dnf install libssl.so.1.1` from a random google search)  

If you're literally running `bash lapce` that would never work: `bash` *directly* takes in bash scripts. If you're typing into the cli/terminal already then `./lapce` is the right way to run the program in the current folder. From the first error saying anything at all, lapce is already an exectuable, so you just need to install libssl and run lapce.

**Reply by mattw-webcv** [reply]
Okay

[root@fedoragamer Lapce]# dnf search libssl
Last metadata expiration check: 0:46:22 ago on Sat 30 Mar 2024 01:06:54 PM AEST.
No matches found.

https://packages.fedoraproject.org/search?query=libssl
"Did you mean libssh?"

[root@fedoragamer Lapce]# dnf -y install openssl
Last metadata expiration check: 0:46:30 ago on Sat 30 Mar 2024 01:06:54 PM AEST.
Package openssl-1:3.1.1-4.fc39.x86_64 is already installed.
Dependencies resolved.
Nothing to do.
Complete!

**Reply by mattw-webcv** [reply]
Maybe various distros organise dependancies differently?

https://packages.fedoraproject.org/search?query=openssl
Is there anything in the above I'd need?

**Reply by MinusGix** [reply]
Either try the command I gave, or maybe try `openssl1.1` package... I don't know all the specifics of fedora dependencies.

**Reply by mattw-webcv** [reply]
Failed to run, cli error now is:
./lapce: error while loading shared libraries: libssl.so.1.1: cannot open shared object file: No such file or directory

I will keep looking in my own time, for now I can report the following:


A dnf install did successfully install libssl.so.1.1, and openssl1.1
A dnf search does not find libssl.so.1.1

A search for (dnf whatprovides libssl.so.1.1)
Only returns "openssl1.1-1:1.1.1q-5.fc39.i686"

I have not done:
Installed openssl1.1-devel
Reports a conflict with the installed openssl-devel

I have done:
reinstalled opennssl1.1
An upgrade/update and rebooted.
Attempted to run lapce as sudo
I've also done various searches for openssl and its variations

Other notes:
Searched forums, the only obvious answer I can see comes from installing dependencies after installing the main package.
This changes up the symlinks and causes problems.

https://www.linode.com/community/questions/22943/how-do-i-fix-error-while-loading-shared-libraries-libsslso11
Details:
cd /usr/lib
ls -l libssl*

My system returns:
[root@fedoragamer lib]# ls -l libssl*
-rwxr-xr-x. 1 root root         486684 Feb 17 10:00     libssl3.so
lrwxrwxrwx. 1 root root     16 Jul 20  2023                 libssl.so.1.1 -> libssl.so.1.1.1q
-rwxr-xr-x. 1 root root         636928 Jul 20  2023       libssl.so.1.1.1q
lrwxrwxrwx. 1 root root     15 Aug 31  2023              libssl.so.3 -> libssl.so.3.1.1
-rwxr-xr-x. 1 root root         685196 Aug 31  2023    libssl.so.3.1.1

Potentially useful posts I do not understand
https://github.com/openssl/openssl/issues/3993
https://ngelinux.com/solved-error-while-loading-shared-libraries-libssl-so-1-1-cannot-open-shared-object-file-no-such-file-or-directory/

**Reply by panekj** [reply]
Im still working on fedora native packages and fixing the libssl1.1 issue in generic build, so I would recommend flatpak package for a time being

**Reply by mattw-webcv** [reply]
I assumed as much, no worries. If there's some way for me to be helpful please let me know.

**Reply by panekj** [reply]
A binary artefact with fixed openssl should be available in next 2-3 hours

**Reply by mattw-webcv** [reply]
Success. Brilliant; thank you.
